### ISSUE TITLE
[ENH] Apply increment/diff in WindowSummarizer 

### ISSUE THREAD

**Issue by TeoSkondras** [sktime/sktime#4959]
I am having a problem applying differencer in my forecasting pipeline for specific columns.

I would like a new lag feature inside the kwargs of windowsummarizer called diff to execute what I just described.

Or add an attribute on Differencer to be able to apply this transformator to specific columns.

**Comment by fkiraly** [comment]
> Or add an attribute on Differencer to be able to apply this transformator to specific columns.

> Is there a way to do that currently?

Yes, we tend to favour composition over adding functionality to individual transformers (as the latter would lead to "mega-feature-transformers" that hare hard to use or maintain").

To apply a transformer such as `Differencer` to only one column, use the `ColumnEnsembleTransformer`, or the square bracket syntax `my_differencer[colname]`, which applies `my_differencer` only to the column(s) `colname`.

A choice to be aware about is what you want to do with the remaining columns, i.e., keep them or drop them. In `ColumnEnsembleTransformer`, you would control this via the `remaider` argumet, i the square bracket syntax you would pass them via `Id`.

For examples, see the tutorial `03_transformers`, or the recent tutorial at EuroPython:
https://github.com/sktime/sktime-tutorial-europython-2023

Kindly let us know if this solves your use case!
